plugin/core/windows.py

```diff
@@ -464,6 +464,14 @@ def _apply_workspace_edit(self, params: 'Dict[str, Any]', client: Client, reques
         # reconstruct/get the actual Client object back. Maybe we can (ab)use our homebrew event system for this?
         client.send_response(Response(request_id, {"applied": True}))
 
+    def _get_session_config(self, params: 'Dict[str, Any]', session: Session, client: Client, request_id: int) -> None:
+        items = []  # type: List[Any]
+        requested_items = params.get("items") or []
+        for requested_item in requested_items:
+            items.append(session.config.settings)
+
+        client.send_response(Response(request_id, items))
```

Comment:
Would it be a lot more work to do it "properly" from the start and expose a hook that LSP plugins (`LanguageHandler` implementations) could react to?

I don't know if it would be really worth it, as with static configuration it will probably be good enough for 99% of the cases, but just in case... :)

Reply:
Right, it would be nice if language handlers can replace this built-in implementation. Would it not be valuable for LSP to provide a functional example for other plugins to improve on (with better defaults for example) ?

Reply:
I guess you are right. Providing default with a way to override would be the best.

For Eslint server, response can include properties like `workspaceFolder` and `workingDirectory`. To provide those, we'd need to be able to override response in `LanguageHandler` but to be fair, I've been running without those two and things are generally working without them. So I'm not sure how important are those.